bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/AbstractExpression.java

```diff
@@ -207,22 +207,51 @@ protected void prune() {
 
     @Override
     public Date getTimeAfter(Date afterTime) {
+
+        Date currentStartDate = getStartDate();
+
+        if (hasFloatingStartDate()) {
+            try {
+                clearCandidates();
+                setStartDate(afterTime);
+            } catch (IllegalArgumentException | ParseException e) {
+                logger.error("An exception occurred while setting the start date : '{}'", e.getMessage());
+            }
+        }
+
         if (getCandidates().isEmpty()) {
```

Comment:
> It should be called again

Does this mean if `setStartDate(afterTime);` results in an empty candidate list another call to `setStartDate(afterTime)` (using the same aster time) can result in an non empty candidate list? Or why should it be called again?